mesonpy/__init__.py

```diff
@@ -815,21 +881,46 @@ def version(self) -> str:
         assert isinstance(version, str)
         return version
 
-    @cached_property
-    def metadata(self) -> bytes:
+    @property
+    def top_level_modules(self) -> Collection[str]:
+        return {
+            os.path.dirname(self._copy_files[file])
```

Comment:
This line doesn't look correct. 

IIUC, we need to normpath, and then replace "/" with ".". Also, the paths have to be relative to the destination anchor, I think.

Reply:
This was the last thing I did yesterday, it was indeed broken, I re-wrote it with slightly better logic.